DOC-2327 TinyMCE 7.0 documentation clean up (#3165)
tinymce/tinymce-docs: modules/ROOT/pages/7.0-release-notes.adoc

In {productname} 7.0 addresses this, by removing the `aria-placeholder` attribute from the editor body. This solution ensures that the editor is accessible to all users.

=== Optional Tooltip for Dialog Footer Toggle Buttons
// #TINY-10672

Previously, the `DialogFooterToggleButton` component required a tooltip property. This meant including a tooltip for all toggle buttons displayed in dialog footers, even if the information wasn't necessary.

As of {productname} 7.0, the tooltip property is now optional for `DialogFooterToggleButton`. This flexibility allows developers to omit the tooltip for buttons where it's not required, providing a cleaner and more focused user experience.

=== Changed the `media_url_resolver` option to use promises.
// #TINY-9154

> // TODO

Author: - [ ] Release note added for TINY-9154

Author: - [ ] add demo to support release note entry

> // TODO

Author: @ArvinJ-H Need a release note entry for TINY-9154.
You can make a commit directly to this ticket.

Author: - [ ] Release note added for TINY-9154 by @ArvinJ-H

In {productname} 7.0 addresses this, by removing the `aria-placeholder` attribute from the editor body. This solution ensures that the editor is accessible to all users.

=== Optional Tooltip for Dialog Footer Toggle Buttons
// #TINY-10672

Previously, the `DialogFooterToggleButton` component required a tooltip property. This meant including a tooltip for all toggle buttons displayed in dialog footers, even if the information wasn't necessary.

As of {productname} 7.0, the tooltip property is now optional for `DialogFooterToggleButton`. This flexibility allows developers to omit the tooltip for buttons where it's not required, providing a cleaner and more focused user experience.

=== Changed the `media_url_resolver` option to use promises.
// #TINY-9154

> In {productname} 7.0, the `media_url_resolver` has been refactored to use promises instead of callbacks. This change brings standardization to the code base and improves error handling.

Author: - [ ] @ArvinJ-H review this please

Reviewer: LGTM

> // TODO: demo to show the function changing from `(data, resolve, reject) => void` to `(data) => Promise<result>`

Author: - [ ] add demo to support TINY-9154

Reviewer 2: outdated, i linked this to the xref:media.adoc#media_url_resolver[media_url_resolver] page, and added entry to the migration guide.

For more information, visit the updated xref:autocompleter.adoc[Autocompleter] documentation.

> === Removed `closeButton` from `NotificationSpec`, close button in notification is now rendered by default.

Author: @ShiridiGandham can you please review this newly added release note entry for TINY-10646?
See preview here: http://docs-feature-70-doc-2327.staging.tiny.cloud/docs/tinymce/latest/7.0-release-notes/#removed-closebutton-from-notificationspec-close-button-in-notification-is-now-rendered-by-default

Author: - [ ] TINY-10646 reviewed by @ShiridiGandham

Reviewer 3: LGTM

[[known-issues]]
== Known issues

This section describes issues that users of {productname} 7.0 may encounter and possible workarounds for these issues.

=== Non-breaking spaces `+&nbsp;+` may be inadvertently inserted around content during Markdown conversion

Currently, Non-breaking spaces `+&nbsp;+` may be inadvertently inserted around content inside `<p>&nbsp;</p>` during Markdown conversion, requiring users to manually remove them post-conversion as the current workaround.

.Example of unexpected insertion of `+&nbsp;+`
[source, html]
----
<p>&nbsp;</p> // unexpected
<h3>Heading</h3>
<p>This is a paragraph.</p>
<ul>
<li>Item 1</li>
<li>Item 2</li>
</ul>
<p>&nbsp;</p> // unexpected
----

=== Tab navigation inside the Enhanced Code Editor currently does not function as expected when the host browser is Safari.
// #TINY-10596

Currently, tab navigation within the Enhanced Code Editor is not working as expected when Safari is the host browser, resulting in a loss of tab functionality.

Currently, there is no workaround for this issue.

> === table that has `colgroup` attribute is highlighted as new in Revision History.

Reviewer 2: ```suggestion
=== Table that has `colgroup` attribute is highlighted as new in Revision History.
```

[[known-issues]]
== Known issues

This section describes issues that users of {productname} 7.0 may encounter and possible workarounds for these issues.

=== Non-breaking spaces `+&nbsp;+` may be inadvertently inserted around content during Markdown conversion

Currently, Non-breaking spaces `+&nbsp;+` may be inadvertently inserted around content inside `<p>&nbsp;</p>` during Markdown conversion, requiring users to manually remove them post-conversion as the current workaround.

.Example of unexpected insertion of `+&nbsp;+`
[source, html]
----
<p>&nbsp;</p> // unexpected
<h3>Heading</h3>
<p>This is a paragraph.</p>
<ul>
<li>Item 1</li>
<li>Item 2</li>
</ul>
<p>&nbsp;</p> // unexpected
----

=== Tab navigation inside the Enhanced Code Editor currently does not function as expected when the host browser is Safari.
// #TINY-10596

Currently, tab navigation within the Enhanced Code Editor is not working as expected when Safari is the host browser, resulting in a loss of tab functionality.

Currently, there is no workaround for this issue.

> === Revision History highlights `table` content with `colgroup` as new.

Reviewer 4: ```suggestion
=== table that has `colgroup` attribute is highlighted as new in Revision History.
```

> TODO//

Reviewer 4: ```suggestion
When a `table` element has `colgroup` attribute, Revision History highlights it as new even though it has no change.
```

Currently, there is no workaround for this issue.

> === Revision History has no integration for saving version.

Reviewer 4: ```suggestion
=== Revision History has no integration for saving a version.
```

> TODO//

Reviewer 4: ```suggestion
```

As a result, the real caret is now hidden when {productname} renders the `FakeCaret`.

=== The `mceTablePasteRowAfter` command, `mceTablePasteColAfter` command, `"Paste row after"` menu item and `"Paste column after"` menu item now place the cursor in the newly pasted row/column to be consistent with the other pasting operations.
// #TINY-6460

Previously, the cursor position was incorrectly calculated after pasting a new row or column when using the `mceTablePasteRowAfter` command, `mceTablePasteColAfter` command, or `"Paste row after"` and `"Paste column after"` UI menu items.

As a consequence, after pasting, the cursor remained in the same position instead of moving to the newly pasted row or column.

{productname} 7.0 addresses this, fixing the cursor calculation.

Now, when using the `mceTablePasteRowAfter` command, `mceTablePasteColAfter` command, or `"Paste row after"` and `"Paste column after"` UI menu items, the cursor is correctly positioned in the newly pasted row or column as expected.

[[known-issues]]
== Known issues

This section describes issues that users of {productname} 7.0 may encounter and possible workarounds for these issues.

> There are two known issues in {productname} 7.0.

Reviewer 2: ```suggestion
There are four known issues in {productname} 7.0.
```

=== Non-breaking spaces `+&nbsp;+` may be inadvertently inserted around content during Markdown conversion

Currently, Non-breaking spaces `+&nbsp;+` may be inadvertently inserted around content inside `<p>&nbsp;</p>` during Markdown conversion, requiring users to manually remove them post-conversion as the current workaround.

.Example of unexpected insertion of `+&nbsp;+`
[source, html]
----
<p>&nbsp;</p> // unexpected
<h3>Heading</h3>
<p>This is a paragraph.</p>
<ul>
<li>Item 1</li>
<li>Item 2</li>
</ul>
<p>&nbsp;</p> // unexpected
----

=== Tab navigation inside the Enhanced Code Editor currently does not function as expected when the host browser is Safari.
// #TINY-10596

Currently, tab navigation within the Enhanced Code Editor is not working as expected when Safari is the host browser and the editor configuration is set to `advcode_inline: true`, resulting in a loss of tab functionality.

.Example of configuration affected.
[source, js]
----
tinymce.init({
  selector: 'textarea', // change this value according to your HTML
  plugins: 'advcode',

>   advcode_inline: true,

Reviewer 2: ```suggestion
```
